src/gc/gc.d

```diff
@@ -551,7 +626,25 @@ struct GC
     }
 
     /**
+     * Request that the GC reallocate a block of memory, attempting to adjust
+     * the size in place if possible. If size is 0, the memory will be freed.
+     *
+     * If p was not allocated by the GC, points inside a block, or is null, no
+     * action will be taken.
+     *
+     * Params:
+     *  p = A pointer to the root of a valid memory block or to null.
+     *  size = The desired allocation size in bytes.
+     *  bits = A bitmask of the attributes to set on this block.
+     *  alloc_size = The actuall size allocated in bytes.
+     *  ti = TypeInfo to describe the memory.
+     *
+     * Returns:
+     *  A reference to the allocated memory on success or null if size is
+     *  zero.  On failure, the original value of p is returned.
```

Comment:
>  On failure, the original value of p is returned.

No. It just crashes if you pass in a non-GC pointer and a size, and throws OutOfMemoryError if allocation fails.